src/dioptra/restapi/db/repository/experiments.py

```diff
@@ -0,0 +1,366 @@
+            self.session, Experiment, resource_ids, deletion_policy
+        )
+
+    def get_one(
+        self,
+        resource_id: int,
+        deletion_policy: utils.DeletionPolicy,
+    ) -> Experiment:
+        """
+        Get the latest snapshot of the given experiment resource; require that
+        exactly one is found, or raise an exception.
+
+        Args:
+            resource_id: A resource ID
+            deletion_policy: Whether to look at deleted experiments, non-deleted
+                experiments, or all experiments
+
+        Returns:
+            An Experiment object
+
+        Raises:
+            EntityDoesNotExistError: if the resource is not found
+        """
+        return utils.get_one_latest_snapshot(
+            self.session, Experiment, resource_id, deletion_policy
+        )
+
+    def get_by_name(
+        self,
+        name: str,
+        group: Group | int,
+        deletion_policy: utils.DeletionPolicy = utils.DeletionPolicy.NOT_DELETED,
+    ) -> Experiment | None:
+        """
+        Get an experiment by name.  This returns the latest version (snapshot)
+        of the experiment.
+
+        Args:
+            name: An experiment name
+            group: A group/group ID, to disambiguate same-named experiments
+                across groups
+            deletion_policy: Whether to look at deleted experiments,
+                non-deleted experiments, or all experiments
+
+        Returns:
+            An experiment, or None if one was not found
+
+        Raises:
+            EntityDoesNotExistError: if the given group does not exist
+            EntityDeletedError: if the given group is deleted
+        """
+        return utils.get_snapshot_by_name(
+            self.session, Experiment, name, group, deletion_policy
+        )
+
+    def get_children(
+        self,
+        experiment: Experiment | int,
+        deletion_policy: utils.DeletionPolicy = utils.DeletionPolicy.NOT_DELETED,
+    ) -> Sequence[EntryPoint]:
+        """
+        Get the child entry points of the given experiment as their latest
+        snapshots.  If an Experiment object is given, the returned sequence
+        may be shorter than len(experiment.children) if any of the children
+        don't exist, or were filtered out due to deletion policy.
+
+        Args:
+            experiment: An Experiment object or resource_id integer primary key
+                value
+            deletion_policy: Whether to look at deleted entry points,
+                non-deleted entry points, or all entry points
+
+        Returns:
+            The entry points
+
+        Raises:
+            EntityDoesNotExistError: if parent does not exist
+            EntityDeletedError: if parent is deleted
+        """
+
+        return utils.get_latest_child_snapshots(
+            self.session,
+            EntryPoint,
+            experiment,
+            deletion_policy,
+        )
+
+    def set_children(
```

Comment:
Later repositories will have more than one available resource dependency to set, and they can be manipulated independently. To support this pattern, we can be explicit about the name of each child in the repository set method, e.g.:

```suggestion
    def set_entrypoint(
```